macosx/QuickLookExtension/PreviewViewController.mm

```diff
@@ -0,0 +1,146 @@
+// This file Copyright © 2010-2022 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "PreviewViewController.h"
+
+#include <libtransmission/error.h>
+#include <libtransmission/torrent-metainfo.h>
+
+#import "NSStringAdditions.h"
+
+#import <QuickLookUI/QLPreviewingController.h>
+
+#pragma mark - TRError
+
+@interface TRError : NSError
+
+- (instancetype)initWith:(tr_error*)error;
+
+@end
+
+@interface TRError ()
+
+@property NSString* errorMessage;
+
+@end
+
+@implementation TRError
+
+- (instancetype)initWith:(tr_error*)error
+{
+    if (self = [super initWithDomain:@"TransmissionError" code:error->code userInfo:nil])
+    {
+        self.errorMessage = @(error->message);
+    }
+    return self;
+}
+
+- (NSString*)localizedDescription
+{
+    return self.errorMessage;
+}
+
+@end
+
+#pragma mark -
+
+@interface PreviewViewController ()<QLPreviewingController>
+
+#pragma mark - IBOutlet
+@property(weak) IBOutlet NSImageView* fileIconImage;
```

Comment:
We keep properties nonatomic across the project. Let's fix it here and in the rest of the file